### PR TITLE
SDIT-1183 int test and fix for edge case

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt [hunk 1]
```diff
@@ -258,7 +258,7 @@ class AdjudicationsService(
         val hearing = adjudicationsApiService.getCharge(
           chargeNumber,
           prisonId,
-        ).reportedAdjudication.hearings.firstOrNull { it.id.toString() == dpsHearingId } ?: throw IllegalStateException(
+        ).reportedAdjudication.hearings.lastOrNull { it.id.toString() == dpsHearingId } ?: throw IllegalStateException(
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt:261]: switching to get latest

File: src/main/resources/application-localstack.yml [hunk 2]
```diff
@@ -35,7 +35,12 @@ hmpps.sqs:
...
+      subscribeFilter: "{\"eventType\":[\"adjudication.report.created\", \"adjudication.damages.updated\", \"adjudication.evidence.updated\", \"adjudication.hearing.created\",\
+       \"adjudication.hearing.updated\", \"adjudication.hearing.deleted\" , \"adjudication.hearingCompleted.created\", \"adjudication.hearingCompleted.deleted\",\
+        \"adjudication.hearingReferral.created\", \"adjudication.hearingReferral.deleted\", \"adjudication.hearingAdjourn.created\", \"adjudication.hearingAdjourn.deleted\",\
+         \"adjudication.punishments.created\", \"adjudication.punishments.updated\", \"adjudication.referral.outcome.prosecution\", \"adjudication.referral.outcome.notProceed\",\
+          \"adjudication.referral.outcome.referGov\", \"adjudication.referral.outcome.deleted\", \"adjudication.outcome.referPolice\", \"adjudication.outcome.notProceed\",\
+           \"adjudication.referral.deleted\", \"adjudication.referral.outcome.deleted\", \"adjudication.outcome.notProceed.deleted\", \"adjudication.hearingOutcome.updated\"]}"
```
Comment (lbennett-moj) [inline src/main/resources/application-localstack.yml:43]: formatting v long yaml line

**Review by petergphillips — APPROVED** [review]
LGTM
